controllers/datafoundationclient.go

```diff
@@ -248,3 +275,52 @@ func (r *dataFoundationClientReconciler) reconcileCSIKMSConnectionDetailsConfigM
 
 	return nil
 }
+
+func (r *dataFoundationClientReconciler) hasOCSVolumes() (bool, error) {
+	storageClassList := &storagev1.StorageClassList{}
+	if err := r.list(storageClassList); err != nil {
+		return false, fmt.Errorf("failed to list Storage Classes: %v", err)
+	}
+	ocsStorageClass := map[string]bool{}
+	rbdProvisionerName := r.offering.Namespace + ocsRBDProvisionerSuffix
+	fsProvisionerName := r.offering.Namespace + ocsCephFsProvisionerSuffix
+	for i := range storageClassList.Items {
+		storageClass := storageClassList.Items[i]
```

Comment:
To prevent a copy operation, please take a reference
```suggestion
		storageClass := &storageClassList.Items[i]
```